lib/guardian/plug/ensure_permissions.ex

```diff
@@ -63,19 +71,38 @@ defmodule Guardian.Plug.EnsurePermissions do
     case Guardian.Plug.claims(conn, key) do
       {:ok, claims} ->
         perms = Map.get(opts, :perms, %{})
-        result = Enum.all?(Map.get(opts, :perm_keys), fn(perm_key) ->
-          found_perms = Guardian.Permissions.from_claims(claims, perm_key)
-          Guardian.Permissions.all?(
-            found_perms,
-            Map.get(perms, perm_key),
-            perm_key
-          )
-        end)
+        result =
```

Comment:
You're right it should be private. It would be matched when no sets/permissions were declared. An unlikely case but possible.